flamingpy/utils/viz.py

```diff
@@ -277,36 +347,48 @@ def draw_EGraph(
         message = "{} at {} node(s) have not yet been computed."
         print(message.format(name.lower(), n_uncomputed))
 
-    # Plotting edges.
-    for edge in egraph.edges:
+    # Plotting nodes legend
+    if isinstance(color_nodes, tuple) and legend:
+        # this two lines are just a handy way to create a legend for
+        # the node colors and attributes by generating handles of empty lines
+        # with the label and color of the node property
+        handles = [
+            mlines.Line2D([], [], marker="o", linewidth=0, color=color, label=node_property)
+            for node_property, color in color_nodes[1].items()
+        ]
+        ax.legend(handles=handles)
 
-        # Color edges based on color_edges if string, or based on
-        # color attribute if True; black otherwise.
-        if isinstance(color_edges, str):
-            color = color_edges
-        elif isinstance(color_edges, dict):
-            color = color_edges.get(egraph.edges[edge].get("weight"))
-        else:
-            color = egraph.edges[edge].get("color") if color_edges else "grey"
+    return ax
 
-        x1, z1, y1 = edge[0]
-        x2, z2, y2 = edge[1]
-        plt.plot([x1, x2], [y1, y2], [z1, z2], color=color, linewidth=0.5)
 
-    if color_nodes == "state" and legend:
-        ax.legend(handles=handles)
+def _get_node_color(egraph, color_nodes, point):
+    """Color nodes based on ``color_nodes`` arg:
 
-    plt.xticks(range(0, 2 * xmax))
-    plt.yticks(range(0, 2 * zmax))
-    ax.set_zticks(range(0, 2 * ymax))
-    ax.set_xlabel("x", labelpad=15)
-    ax.set_ylabel("z", labelpad=15)
-    ax.set_zlabel("y", labelpad=15)
-    if not display_axes:
-        ax.axis("off")
-    plt.tight_layout(pad=5)
-    plt.draw()
-    return fig, ax
+    - if `color_nodes` is a string use the string as color,
+    - using the attribute and color dict if `color_nodes` is a tuple(str,dict),
+    - or based on color attribute (when available) if `color_nodes` is bool and True;
+    - black otherwise.
+    """
+    if isinstance(color_nodes, str):
+        color = color_nodes
+    elif isinstance(color_nodes, tuple):
+        node_attribute, color_dict = color_nodes
+        if not (isinstance(node_attribute, str) and isinstance(color_dict, dict)):
+            raise ValueError(
+                "Inappropiate value for `color_nodes` argument:"
+                "Check it complies with the type `tuple(str, dict)`"
```

Comment:
```suggestion
                "Check that it complies with the type `tuple(str, dict)`,"
```
This may also change pending discussion above.